xla/service/gpu/gpu_windowed_einsum_handler.cc

```diff
@@ -48,16 +48,15 @@ namespace m = match;
 // and type conversions of FP8 operands into the bodies of their while loops,
 // i.e. rewrites
 //
-//   inputs --> dequant --> while loop {...}
+//   inputs --> dequant --> while loop {collective-permute/dot/etc}
 //
 // into
 //
-//   inputs --> while loop {dequant --> ...}.
-absl::StatusOr<bool> ShiftDequantizationF8(HloInstruction* instr) {
-  HloComputation* comp = instr->parent();
+//   inputs --> while loop {dequant --> collective-permute/dot/etc}.
+absl::Status ShiftDequantizationF8(HloComputation* comp) {
   HloInstruction* while_instr = comp->WhileCallInstruction();
-  if (!while_instr) {
-    return false;
+  if (!while_instr || while_instr->operand(0)->user_count() != 1) {
```

Comment:
Comment why there is the check `while_instr->operand(0)->user_count() != 1`. E.g.:

The while-loop tuple operand must only be used by the while-loop, since we modify the tuple arguments